hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/NegativeBurnContractSuite.java

```diff
@@ -0,0 +1,269 @@
+import com.hedera.services.bdd.spec.keys.KeyShape;
+import com.hedera.services.bdd.suites.HapiSuite;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Tag;
+
+@HapiTestSuite
+@Tag(SMART_CONTRACT)
+public class NegativeBurnContractSuite extends HapiSuite {
+    private static final Logger log = LogManager.getLogger(NegativeBurnContractSuite.class);
+    public static final String CONTRACT = "NegativeBurnContract";
+    private static final long GAS_TO_OFFER = 4_000_000L;
+    private static final String NFT = "NFT";
+    private static final String FIRST = "First!";
+    private static final String SECOND = "Second!";
+    private static final String CONTRACT_KEY = "ContractKey";
+
+    public static void main(String... args) {
+        new NegativeBurnContractSuite().runSuiteAsync();
+    }
+
+    @Override
+    public boolean canRunConcurrent() {
+        return true;
+    }
+
+    @Override
+    protected Logger getResultsLogger() {
+        return log;
+    }
+
+    @Override
+    public List<HapiSpec> getSpecsInSuite() {
+        return List.of(burnWithNegativeAmount());
+    }
+
+    @HapiTest
+    final HapiSpec burnWithNegativeAmount() {
+        final var negativeBurnFungible = "negativeBurnFungible";
+        final var negativeBurnNFT = "negativeBurnNFT";
+        final AtomicReference<Address> tokenAddress = new AtomicReference<>();
+        final AtomicReference<Address> nftAddress = new AtomicReference<>();
+        return defaultHapiSpec("burnWithNegativeAmount")
+                .given(
+                        uploadInitCode(CONTRACT),
+                        contractCreate(CONTRACT),
+                        cryptoCreate(TOKEN_TREASURY),
+                        tokenCreate(TOKEN)
+                                .tokenType(FUNGIBLE_COMMON)
+                                .treasury(TOKEN_TREASURY)
+                                .initialSupply(1_000L)
+                                .exposingAddressTo(tokenAddress::set),
+                        tokenCreate(NFT)
+                                .tokenType(NON_FUNGIBLE_UNIQUE)
+                                .treasury(TOKEN_TREASURY)
+                                .initialSupply(0L)
+                                .supplyKey(TOKEN_TREASURY)
+                                .adminKey(TOKEN_TREASURY)
+                                .exposingAddressTo(nftAddress::set),
+                        mintToken(NFT, List.of(copyFromUtf8(FIRST), copyFromUtf8(SECOND))))
+                .when(
+                        sourcing(() -> contractCall(CONTRACT, "burnFungibleNegativeLong", tokenAddress.get())
+                                .gas(GAS_TO_OFFER)
+                                .hasKnownStatus(CONTRACT_REVERT_EXECUTED)
+                                .via(negativeBurnFungible)
+                                .logged()),
```

Comment:
We can add emptyChildRecordsCheck(negativeBurnFungible, CONTRACT_REVERT_EXECUTED)

to assert that there are no childs